src/main/java/org/mskcc/oncokb/curation/repository/AlterationRepository.java

```diff
@@ -42,12 +45,27 @@ public interface AlterationRepository extends JpaRepository<Alteration, Long>, J
 
     @Query(
         "select distinct alteration from Alteration alteration" +
+        " left join fetch alteration.flags" +
         " left join fetch alteration.genes g" +
         " where g.id in (:id) and" +
         " (alteration.name =:query or alteration.alteration=:query )"
     )
     List<Alteration> findByNameOrAlterationAndGenesId(@Param("query") String query, @Param("id") Long geneId);
 
+    @Query(
+        "select distinct alteration from Alteration alteration" +
+        " left join fetch alteration.genes g" +
+        " left join fetch alteration.flags" +
+        " where g.id in :genes and " +
+        " alteration.consequence = :consequence and ((alteration.start <= :end and alteration.end >= :start) or (alteration.end >= :start and alteration.start <= :end)) "
```

Comment:
```suggestion
        " alteration.consequence = :consequence and (alteration.start <= :end and alteration.end >= :start) "
```

This should be sufficient to check if two ranges overlap. The original had two duplicated conditioned with OR.